### DISCUSSION TITLE
Writing yet another git webview -- how to know if a Blob is a LFS entry

### DISCUSSION THREAD

**Discussion by axkibe** [git-lfs/git-lfs#5121] [Q&A] [answered]
I wrote a git server with integrated LFS support in node.js a while ago: https://gitlab.com/csc1/gitengine

Part of it was having a cgit web interface, unfortunally it cannot show LFS entries (quite understandable).

Instead of hacking this into that, I'm writing my own web interface (in node.js with nodegit/libgit). Shouldn't be too hard.

Now when I'm getting a blob from git (via nodegit/libgit) what is the recommended heuristics to know if it's actually a LFS link?

Should I look in the blob content and treat everything that starts with "version https://git-lfs.github.com/spec/v1" as LFS link?

Or should I parse .gitattributes and see if it applies?

Or is something else the recommended method?

**Comment by bk2204** [comment, marked as answer]
Hey,

The way to know if a file is an LFS file is the `.gitattributes` file.  However, be aware that sometimes a file ends up being incorrectly committed as a Git file instead of an LFS file even though `.gitattributes` says that it's supposed to be an LFS file.  In such a case, the file will not parse as a pointer or may be larger than 1024 bytes.  You may want to handle that case gracefully.

**Comment by axkibe** [comment]
Thanks!

I was aware of the caveat, simply I guess when a user not having git-lfs installed commits said file, it ends up as binary. (might be worth having a server pre-commit check to deny these cases tough).

So:
1 query git attributes, see if LFS filter applies (if no -> not LFS)
2. query if blob size > 1024 (no -> not LFS)
3. try to parse the LFS link (if fails -> not LFS)
4. it's a LFS file, get the Blob from my LFS storage. (if not there -> complain bitterly :) )

Edit. actually make 2 before 1, since this is very cheap when having the git blob already (thanks for that input)

**Reply by bk2204** [reply]
Yup, that sounds correct.
